app/src/main/java/com/github/multimatum_team/multimatum/AccountActivity.kt

```diff
@@ -0,0 +1,151 @@
+
+        //Configure sign-in to request user's credential
+        val gso: GoogleSignInOptions = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id)).requestEmail().build()
+        //configure client
+        googleSignInClient = GoogleSignIn.getClient(this, gso)
+        val account = GoogleSignIn.getLastSignedInAccount(this)
+
+        //init firebase auth
+        firebaseAuth = FirebaseAuth.getInstance()
+        checkUser()
+
+
+        //configure button
+        findViewById<SignInButton>(R.id.sign_in_button).visibility = View.VISIBLE
+        findViewById<SignInButton>(R.id.sign_in_button).setOnClickListener{
+            Log.d(TAG, "onCreate begin Google SignIn:")
+            signIn()
+        }
+
+
+    }
+
+    private fun checkUser(){
+        //check if user is logged in or not
+        val firebaseUser = firebaseAuth.currentUser
+        if (firebaseUser != null){
+            //TODO: start profile fragment
+        }
+    }
+
+    private fun signIn(){
+        val signInIntent: Intent = googleSignInClient.signInIntent
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+    @SuppressLint("VisibleForTests")
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // Result returned from launching the Intent from GoogleSignInClient.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            // The Task returned from this call is always completed, no need to attach
+            // a listener.
+            Log.d(TAG, "onActivityResult: Google SignIn intent result")
+
+            val task: Task<GoogleSignInAccount> = GoogleSignIn.getSignedInAccountFromIntent(data)
+            handleSignInResult(task)
+            try{
+                val account = task.getResult(ApiException::class.java)!!
+                Log.d(TAG, "firebaseAuthWithGoogle:"+account.id)
+                firebaseAuthWithGoogleAccount(account)
+            }catch (e: Exception){
+                Log.d(TAG, "onActivityResult: ${e.message}")
+            }
+        }else{
+            Toast.makeText(this@AccountActivity, "Loggin faile due too wrong request Code", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun firebaseAuthWithGoogleAccount(account: GoogleSignInAccount) {
+        Log.d(TAG, "firebaseAuthWithGoogleAccount: begin firebase auth with google account")
+        val credential = GoogleAuthProvider.getCredential(account.idToken, null)
+        firebaseAuth.signInWithCredential(credential).addOnSuccessListener { authResult ->
+            //log in success
+            Log.d(TAG, "firebaseAuthWithGoogleAccount: LoggedIn")
+
+            //get loggedIn user
+            val firebasUser = firebaseAuth.currentUser
+            //get user info
+            val uid = firebasUser!!.uid
+            val email = firebasUser.email
```

Comment:
Maybe check if email are non-null as well.

Reply:
As google account need an email, it can't be null anyway

Reply:
We never know, some corrupted gAccount may try to connect

Reply:
As you have to enter an email to be log in with google, it seems pretty impossible eventfor a corrputed google to don't have an email